### PR TITLE
Move documentation from the docs repo

### PR DESCRIPTION
**Pull request by aslakhellesoy**
# Description

Document Cucumber Expressions in the same repo where their implementation lives.

# Motivation & context

* [Your docs belong in the code repository](https://daniel.haxx.se/blog/2021/09/04/making-world-class-docs-takes-effort/)
* https://github.com/cucumber/docs/issues/633
* https://github.com/cucumber/common/issues/1521

## Update required of cucumber.io/docs

Ref: https://github.com/cucumber/docs/pull/667

# Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](../CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,197 @@
...
+
+The simplest Cucumber Expression that matches that text would be the text itself,
+but we can also write a more generic expression, with an `int` *output parameter*:
+
+    I have {int} cucumbers in my belly
+
+When the text is matched against that expression, the number `42` is extracted
+from the `{int}` output parameter and passed as an argument to the [step definition](https://cucumber.io/docs/cucumber/step-definitions).
+
+The following text would not match the expression:
```
Comment (olleolleolle) [inline README.md:29]: Perhaps bolding the "not"?

File: README.md [hunk 2]
```diff
@@ -1,7 +1,197 @@
...
+
+If we want the `{color}` output parameter to be converted to a `Color` object,
+we can define a custom parameter type in Cucumber's [configuration](https://cucumber.io/docs/cucumber/configuration).
+
+The table below explains the various arguments you can pass when defining
+a parameter type.
+
+| Argument      | Description |
+| ------------- | ----------- |
+| `name`        | The name the parameter type will be recognised by in output parameters.
```
Comment (olleolleolle) [inline README.md:80]: Are we using American or British English as our target language? (US color+recognize vs UK colour+recognise.)

**Review by olleolleolle — COMMENTED** [review]
This is such a good idea.

**Comment by aurelien-reeves** [comment]
What about moving *Grammar* and "Production rules* paragraphs elsewhere?
Those are far more technical, maybe in a CONTRIBUTING.md or ARCHITECTURE.md document?

**Comment by olleolleolle** [comment]
Off-topic, but grammar-related: When this morning I visited the README, I wondered, "this grammar notation, what language is that written in?"

File: README.md [hunk 3]
```diff
@@ -1,7 +1,197 @@
...
+Cucumber Expressions is an alternative to [Regular Expressions](https://en.wikipedia.org/wiki/Regular_expression)
+with a more intuitive syntax.
+
+Cucumber supports both Cucumber Expressions and Regular Expressions for defining
+[Step Definitions](https://cucumber.io/docs/cucumber/step-definitions), but you cannot
+mix Cucumber Expression syntax with Regular Expression syntax in the same expression.
+
+On platforms that don't have a literal syntax for regular expressions (such as Java),
+Cucumber will create a Cucumber Expression by default. To use Regular Expressions, add anchors (starting with `^` and ending with `$`) or forward slashes (`/`). For more information, see
+[Cucumber Expression - Java Heuristics](https://github.com/cucumber/cucumber/blob/master/cucumber-expressions/java/heuristics.adoc).
```
Comment (aurelien-reeves) [inline README.md:12]:
```suggestion
[Cucumber Expression - Java Heuristics](https://github.com/cucumber/cucumber-expressions/blob/main/java/heuristics.adoc).
```

**Review by aurelien-reeves — CHANGES_REQUESTED** [review]

**Comment by mpkorstanje** [comment]
Parts of this documentation - specifically the how-to-do-in-language-x - aren't quite part of this cucumber expressions module. The module only provides a registry. Would it be an idea to link to the relevant sections in the Cucumber implementation read me's for those parts?

**Comment by aslakhellesoy** [comment]
I had the same thought @mpkorstanje and decided to leave here for now since some of the cucumber implementations don’t (yet) have a github markdown page we can link to.

I’d prefer to do that in a second pass, which would involve creating a ticket in all the relevant repos, and wait for these tickets to be resolved.

File: README.md (hunk 2 above)
Comment (aslakhellesoy) [reply README.md:80]: I think British English is the predominant language used in our docs, but I’m sure there are many inconsistencies.

**Comment by aslakhellesoy** [comment]
> Off-topic, but grammar-related: When this morning I visited the README, I wondered, "this grammar notation, what language is that written in?"

It’s pseudo-[EBNF](https://en.wikipedia.org/wiki/Extended_Backus–Naur_form)

**Comment by olleolleolle** [comment]
> > Off-topic, but grammar-related: When this morning I visited the README, I wondered, "this grammar notation, what language is that written in?"
> 
> It’s pseudo-[EBNF](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form)

Perfect. Then this makes it perhaps-more-legible: https://github.com/cucumber/cucumber-expressions/pull/10

**Pull request merged**